app/models/game.rb

```diff
@@ -4,11 +4,67 @@ class Game < ActiveRecord::Base
 
   has_many :pieces
 
+  after_create :populate_board!
+
+  def populate_board!
+    # !! board needs to be rearranged so that a white square is on bottom right (7, 7)
+    # white pieces (bottom)
+    # rook: [0,0],[7,0]; knight: [1,0],[6,0]; bishop: [2,0][5,0]; queen: [3,0]; king[4,0]
+    # pawns: [0->7,1]
+    # black pieces (top)
+    # rook: [0,7],[7,7]; knight: [1,7],[6,7]; bishop: [2,7][5,7]; queen: [3,7]; king[4,7]
+    # pawns: [0->7,7]
+    create_non_pawn_pieces(id, 'white')
+    create_non_pawn_pieces(id, 'black')
+    create_pawn_pieces(id, 'white')
+    create_pawn_pieces(id, 'black')
+  end
+
   def board_state
     board = []
     pieces.each do |piece|
       board << [piece.x_coordinate, piece.y_coordinate]
     end
     board
   end
+
+  private
+
+  def create_non_pawn_pieces(id, color)
+    non_pawn_pieces = [
+      Rook.create(x_coordinate: 0),
```

Comment:
This is creating a new database object and then later updating it (2 writes to the database).

Another way of doing this would be like...

```
non_pawn_pieces = [
  pieces.new(x_coordinate: 0, type: 'Rook'),
  ..ect...
]
```

^ This code would automatically be setting `game_id` for you, because we're taking advantage of the existing `has_many pieces` association.

Later can do
```
piece.update_attributes(y_coordinate: 0, color: color, captured: false)
```